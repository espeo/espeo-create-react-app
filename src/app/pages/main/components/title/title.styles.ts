import styled from 'styled-components';

export const Title = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: ${props => props.theme.colors.violet};
`;
