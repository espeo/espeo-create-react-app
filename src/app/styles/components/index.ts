export * from './buttons';
export * from './containers';
export * from './image';
export * from './text';
