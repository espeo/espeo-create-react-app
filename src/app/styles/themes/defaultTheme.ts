import baseStyled, { ThemedStyledInterface } from 'styled-components';

export const defaultTheme = {
  colors: {
    dark: '#222',
    light: '#fff',
    green: '#080',
    red: '#f00',
    black: '#000',
    violet: '#db7093',
    darkViolet: '#512B37',
    brown: '#996B61',
    gray: '#9D9DB9',
    lightGray: '#E9EBF1',
    nude: '#EEDDDC',
    peach: '#F2C39C',
  },
};

export type Theme = typeof defaultTheme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
