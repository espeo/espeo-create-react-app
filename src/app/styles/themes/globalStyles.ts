import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    padding: 0;
    margin: 0;
  }
`;
