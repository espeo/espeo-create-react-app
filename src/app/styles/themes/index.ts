export { Theme, styled, defaultTheme } from './defaultTheme';

export { GlobalStyles } from './globalStyles';
