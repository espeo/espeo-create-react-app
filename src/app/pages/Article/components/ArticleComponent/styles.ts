import { styled } from '@core/styles/themes/defaultTheme';
import { Link } from 'react-router-dom';

export const ArticleContainer = styled.div`
  width: 50%;
  margin: 3% auto;
`;

export const BackLink = styled(Link)`
  text-decoration: none;
`;

export const BackLinkText = styled.p`
  font-size: 14px;
  font-weight: bold;
  color: ${props => props.theme.colors.darkViolet};
  font-family: Arial;
`;

export const BackImage = styled.div``;

export const SourceLink = styled.a`
  font-family: Georgia;
  font-size: 14px;
  padding-right: 8px;
  color: ${props => props.theme.colors.darkViolet};
  text-decoration: none;
  &:visited {
    color: blue;
  }
`;
