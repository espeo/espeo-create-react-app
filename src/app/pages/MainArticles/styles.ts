import { styled } from '@styles/themes/defaultTheme';

export const Select = styled.select`
  height: 20px;
  width: 80px;
  margin-right: 2%;
  border-radius: 4px;
  border-color: ${props => props.theme.colors.darkViolet};
`;

export const ArticlesWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: center;
  margin-top: 3%;
`;

export const ArticleWrapper = styled.div`
  width: 29%;
  padding-right: 2%;
  padding-left: 2%;
  padding-top: 2%;
`;
export const FilterButtonWrapper = styled.div`
  padding-left: 2%;
`;
export const FiltersWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding-top: 2%;
`;

export const FilterWrapper = styled.div`
  display: flex;
  margin-right: 0.5%;
`;
