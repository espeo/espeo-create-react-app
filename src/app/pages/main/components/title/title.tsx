import React from 'react';
import { Title as TitleWrapper } from './title.styles';

interface TitleProps {
  text: string;
}

export const Title = ({ text }: TitleProps) => (
  <TitleWrapper>{text}</TitleWrapper>
);
