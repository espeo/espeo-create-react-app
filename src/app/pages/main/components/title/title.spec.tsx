import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { shallowWithTheme, withThemeProvider } from '@core/tests/theme-helpers';
import { defaultTheme } from '@core/styles/themes';

import { Title } from './title';

describe('Main Title Component', () => {
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    component = renderer.create(
      withThemeProvider(<Title text="title" />, defaultTheme),
    );
  });

  test('should be defined', () => {
    expect(component).toBeDefined();
  });

  test('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have proper styles', () => {
    const tree = component.toJSON();
    expect(tree).toHaveStyleRule('color', defaultTheme.colors.violet);
  });

  test('should render string from title attribute', () => {
    const text = shallowWithTheme(<Title text="title" />, defaultTheme)
      .children()
      .text();
    expect(text).toEqual('title');
  });
});
