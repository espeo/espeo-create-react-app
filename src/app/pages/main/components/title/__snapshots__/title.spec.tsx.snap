// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Title Component should match snapshot 1`] = `
.c0 {
  font-size: 1.5em;
  text-align: center;
  color: #db7093;
}

<h1
  className="c0"
>
  title
</h1>
`;
